lib/solidus_gdpr/data_segments/profile_segment.rb

```diff
@@ -8,28 +8,29 @@ class ProfileSegment < Base
       #
       # @return [Hash]
       def export
-        serialize(user, with: :profile)
+        serialize(user, with: :profile) if user
```

Comment:
It's probably better to return `nil` in the serializer itself. That's what we do for other models such as `Spree::Address`.